internal/charmhub/download.go

```diff
@@ -243,36 +207,23 @@ func (c *DownloadClient) download(ctx context.Context, resourceURL *url.URL, arc
 		progressBar = opts.progressBar
 	}
 
-	hasher := c.getHasher(opts.digestType)
+	hasher256 := sha256.New()
+	hasher384 := sha512.New384()
 
-	writer := io.MultiWriter(f, hasher, progressBar)
-	size, err := io.Copy(writer, r.Body)
+	size, err := io.Copy(f, io.TeeReader(r.Body, io.MultiWriter(hasher256, hasher384, progressBar)))
 	if err != nil {
 		return nil, errors.Trace(err)
 	} else if size != r.ContentLength {
 		return nil, errors.Errorf("downloaded size %d does not match expected size %d", size, r.ContentLength)
 	}
 
-	digest := hex.EncodeToString(hasher.Sum(nil))
-
 	return &Digest{
-		DigestType: opts.digestType,
-		Hash:       digest,
-		Size:       size,
+		SHA256: fmt.Sprintf("%x", hasher256.Sum(nil)),
+		SHA384: hex.EncodeToString(hasher384.Sum(nil)),
```

Comment:
Can these be done consistently please? `hex.EncodeToString` works for sha256 as well.

Reply:
I think so